reconcile/gabi_authorized_users.py

```diff
@@ -27,15 +29,28 @@
 
 QONTRACT_INTEGRATION = "gabi-authorized-users"
 QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
-EXPIRATION_MAX = 90
+EXPIRATION_DAYS_MAX = 90
 
 
-def construct_gabi_oc_resource(name: str, users: Iterable[str]) -> OpenshiftResource:
+def construct_gabi_oc_resource(
+    name: str, expiration_date: date, users: Iterable[str]
+) -> OpenshiftResource:
+    # Support the legacy users file. To be removed in the future.
+    _users = users if expiration_date >= date.today() else []
     body = {
         "apiVersion": "v1",
         "kind": "ConfigMap",
         "metadata": {"name": name, "annotations": {"qontract.recycle": "true"}},
-        "data": {"authorized-users.yaml": "\n".join(users)},
+        "data": {
+            "authorized-users.yaml": "\n".join(_users),
```

Comment:
**To the reviewer:**

The old legacy users file support has been kept back for backward compatibility.  The current and older version of GABI should work without problems, especially as the new version supports legacy users and the new JSON-based configuration files.